### PR TITLE
[ROCm] prefer hip interfaces over roc during hipify

### PR DESCRIPTION
**Pull request by jeffdaily**
### Description
Change the hipify step to remove the -roc option to hipify-perl.  This will prefer hipblas over rocblas.  rocblas can still be called directly such as in TunableOp.

### Motivation and Context
hip interfaces are preferred over roc for porting from cuda to hip.  Calling roc interfaces is meant for ROCm-specific enhancements or extensions.

### CODE REVIEW

File: onnxruntime/core/providers/rocm/rocm_call.cc [hunk 1]
```diff
@@ -135,6 +135,8 @@ std::conditional_t<THRW, void, Status> RocmCall(
 
 template Status RocmCall<hipError_t, false>(hipError_t retCode, const char* exprString, const char* libName, hipError_t successCode, const char* msg, const char* file, const int line);
 template void RocmCall<hipError_t, true>(hipError_t retCode, const char* exprString, const char* libName, hipError_t successCode, const char* msg, const char* file, const int line);
+template Status RocmCall<hipblasStatus_t, false>(hipblasStatus_t retCode, const char* exprString, const char* libName, hipblasStatus_t successCode, const char* msg, const char* file, const int line);
+template void RocmCall<hipblasStatus_t, true>(hipblasStatus_t retCode, const char* exprString, const char* libName, hipblasStatus_t successCode, const char* msg, const char* file, const int line);
 template Status RocmCall<rocblas_status, false>(rocblas_status retCode, const char* exprString, const char* libName, rocblas_status successCode, const char* msg, const char* file, const int line);
```
Comment (TedThemistokleous) [inline onnxruntime/core/providers/rocm/rocm_call.cc:140]: Why keep the only template still? Backwards compatibility?

File: onnxruntime/core/providers/rocm/shared_inc/rocm_call.h [hunk 2]
```diff
@@ -17,6 +17,7 @@ std::conditional_t<THRW, void, Status> RocmCall(
 
 #define HIP_CALL(expr) (RocmCall<hipError_t, false>((expr), #expr, "HIP", hipSuccess, "", __FILE__, __LINE__))
 #define ROCBLAS_CALL(expr) (RocmCall<rocblas_status, false>((expr), #expr, "ROCBLAS", rocblas_status_success, "", __FILE__, __LINE__))
```
Comment (TedThemistokleous) [inline onnxruntime/core/providers/rocm/shared_inc/rocm_call.h:19]: Same with here. Why not remove

**Review by TedThemistokleous — COMMENTED** [review]
Guessing we're keeping the ROCBLAS_CALL and other templates/defines to ensure if someone tries to build the older version support is still there?

**Comment by tianleiwu** [comment]
Python format failed. Please follow https://github.com/microsoft/onnxruntime/blob/main/docs/Coding_Conventions_and_Standards.md#linting to fix

File: onnxruntime/core/providers/rocm/rocm_call.cc (hunk 1 above)
Comment (jeffdaily) [reply onnxruntime/core/providers/rocm/rocm_call.cc:140]: TunableOp still uses rocblas directly.

File: onnxruntime/core/providers/rocm/shared_inc/rocm_call.h (hunk 2 above)
Comment (jeffdaily) [reply onnxruntime/core/providers/rocm/shared_inc/rocm_call.h:19]: TunableOp still uses rocblas directly.  Also kernel explorer module.

**Comment by jeffdaily** [comment]
> Python format failed. Please follow https://github.com/microsoft/onnxruntime/blob/main/docs/Coding_Conventions_and_Standards.md#linting to fix

@tianleiwu Done.

**Review by TedThemistokleous — APPROVED** [review]
Understood. Thanks for answering questions. LGTM

**Review by tianleiwu — APPROVED** [review]

**Comment by tianleiwu** [comment]
Please take a look at the unit test failure in ROCm pipeline:
https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1521002&view=logs&j=f2f63060-d9d6-52d0-adee-b97db5a9ab91&t=28e21ca6-87a4-5e1e-0441-72b5e8326f2d&l=8852

```
 [ RUN      ] MatmulIntegerOpTest.MatMulInteger_int8_t
1: 2024-10-11 20:32:23.974607957 [E:onnxruntime:Default, rocm_call.cc:126 RocmCall] 
1: 2024-10-11 20:32:23.974636057 [E:onnxruntime:MatMulInteger, sequential_executor.cc:516 ExecuteKernel] Non-zero status code returned while running MatMulInteger node. Name:'node1' Status Message: 
1: /onnxruntime_src/onnxruntime/test/providers/base_tester.cc:337: Failure
1: Expected equality of these values:
1:   expect_result
1:     Which is: 4-byte object <00-00 00-00>
1:   ExpectResult::kExpectFailure
1:     Which is: 4-byte object <01-00 00-00>
1: Run failed but expected success: Non-zero status code returned while running MatMulInteger node. Name:'node1' Status 
```
Is it int

**Comment by jeffdaily** [comment]
> @jeffdaily, Please take a look at the unit test failure in ROCm pipeline: https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1521002&view=logs&j=f2f63060-d9d6-52d0-adee-b97db5a9ab91&t=28e21ca6-87a4-5e1e-0441-72b5e8326f2d&l=8852
> 
> ```
>  [ RUN      ] MatmulIntegerOpTest.MatMulInteger_int8_t
> 1: 2024-10-11 20:32:23.974607957 [E:onnxruntime:Default, rocm_call.cc:126 RocmCall] 
> 1: 2024-10-11 20:32:23.974636057 [E:onnxruntime:MatMulInteger, sequential_executor.cc:516 ExecuteKernel] Non-zero status code returned while running MatMulInteger node. Name:'node1' Status Message: 
> 1: /onnxruntime_src/onnxruntime/test/providers/base_tester.cc:337: Failure
> 1: Expected equality of these values:
> 1:   expect_result
> 1:     Which is: 4-byte object <00-00 00-00>
> 1:   ExpectResult::kExpectFailure
> 1:     Which is: 4-byte object <01-00 00-00>
> 1: Run failed but expected success: Non-zero status code returned while running MatMulInteger node. Name:'node1' Status 
> ```

@tianleiwu thanks for pointing this out.  Bug fixed by latest commit.

**Review by tianleiwu — APPROVED** [review]

**Pull request merged**